utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,452 @@
+        assert (
+            False
+        ), f"Raised Exception in {model_format} model registration/deployment: {e}"
+
+    # 2.) Check model status
+    try:
+        ml_model_status = ml_client.get_model_info(model_id)
+        print()
+        print("Model Status:")
+        print(ml_model_status)
+        assert ml_model_status.get("model_format") == model_format
+        assert ml_model_status.get("algorithm") == "TEXT_EMBEDDING"
+    except Exception as e:
+        assert False, f"Raised Exception in getting {model_format} model info: {e}"
+
+    # 3.) Generate embeddings
+    try:
+        embedding_output = ml_client.generate_embedding(model_id, TEST_SENTENCES)
+        assert len(embedding_output.get("inference_results")) == 2
+        embedding_data = [
+            embedding_output["inference_results"][i]["output"][0]["data"]
+            for i in range(len(TEST_SENTENCES))
+        ]
+    except Exception as e:
+        assert (
+            False
+        ), f"Raised Exception in generating sentence embedding with {model_format} model: {e}"
+
+    # 4.) Undeploy the model
+    try:
+        ml_client.undeploy_model(model_id)
+        ml_model_status = ml_client.get_model_info(model_id)
+        assert ml_model_status.get("model_state") != "UNDEPLOY_FAILED"
+    except Exception as e:
+        assert False, f"Raised Exception in {model_format} model undeployment: {e}"
+
+    # 5.) Delete the model
+    try:
+        delete_model_obj = ml_client.delete_model(model_id)
+        assert delete_model_obj.get("result") == "deleted"
+    except Exception as e:
+        assert False, f"Raised Exception in deleting {model_format} model: {e}"
+
+    # 6.) Return embedding outputs for model verification
+    return embedding_data
+
+
+def verify_embedding_data(
+    original_embedding_data: List["DTypeLike"],
+    tracing_embedding_data: List["DTypeLike"],
+    model_format: str,
+) -> None:
+    """
+    Verify the embeddings generated by the traced model with that of original model
+
+    :param original_embedding_data: Embedding outputs of TEST_SENTENCES generated by the original model
+    :type original_embedding_data: List['DTypeLike']
+    :param tracing_embedding_data: Embedding outputs of TEST_SENTENCES generated by the traced model
+    :type tracing_embedding_data: List['DTypeLike']
+    :param model_format: Model format ("TORCH_SCRIPT" or "ONNX")
+    :type model_format: string
+    """
+    failed_cases = []
+    for i in range(len(TEST_SENTENCES)):
+        try:
+            np.testing.assert_allclose(
+                original_embedding_data[i],
+                tracing_embedding_data[i],
+                rtol=RTOL_TEST,
+                atol=ATOL_TEST,
+            )
+        except Exception as e:
+            failed_cases.append((TEST_SENTENCES[i], e))
+
+    if len(failed_cases):
+        print()
+        print(
+            f"Original embeddings DOES NOT matches {model_format} embeddings in the following case(s):"
```

Comment:
I think i would like to add the model id information in the output as well. In that case, rather than sending `model_format` and `model_id` in this function, we can just send true/false from this function as return value and then handle the exception from the parent main function.

Reply:
Sure!